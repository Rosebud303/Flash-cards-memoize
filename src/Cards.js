import React, { Component } from 'react';
import Description from './Description.js'
import './_Cards.scss'

export default class Cards extends Component {
    constructor(props){
        super(props)
        this.state = {
            showDescription: false,
            disableCard: this.props.reset
        }
    }

    toggleDescription = () => {
        this.setState({
            showDescription: !this.state.showDescription,
            disableCard: true
        })
    }

    checkType = (e) => {
        e.preventDefault();
        if(this.props.proto.type === e.target.value && this.props.wrong.includes(this.props.proto.id)) {
            let idx = this.props.wrong.indexOf(this.props.proto.id)
            this.props.wrong.splice(idx, 1)
            localStorage.setItem('wrongAnswers', JSON.stringify(this.props.wrong))
        } else if (this.props.proto.type !== e.target.value && !this.props.wrong.includes(this.props.proto.id)) {
            this.props.wrong.push(this.props.proto.id)
            localStorage.setItem('wrongAnswers', JSON.stringify(this.props.wrong))
        }
        this.toggleDescription();
    }

    render() {
        if(!this.state.showDescription){
            return(
                <input className='unclicked'onClick={this.toggleDescription} disabled={this.state.disableCard} value={this.props.proto.name}>
                </input>
            )
        } else {
            return(
                <Description proto={this.props.proto} 
                    check={this.checkType}
                />
            )
        }

    }
}