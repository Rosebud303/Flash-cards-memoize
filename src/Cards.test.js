import React from 'react';
import Cards from './Cards.js';
import { shallow } from 'enzyme';

const mockArray = [{
    "id": 1,
    "name": ".concat",
    "definition": "The concat() method is used to merge two or more arrays. This method does not change the existing arrays, but instead returns a new array.",
    "type": "accessor"
  },
  {
    "id": 2,
    "name": ".copyWithin",
    "definition": "The copyWithin() method shallow copies part of an array to another location in the same array and returns it, without modifying its size.",
    "type": "mutator"
  },
  {
    "id": 3,
    "name": ".entires",
    "definition": "The entries() method returns a new Array Iterator object that contains the key/value pairs for each index in the array.",
    "type": "iteration"
  }
]
 
const mockObj = {"id": 2};
const mockFalse = false;


describe('Cards', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = shallow (
            <Cards key={mockObj.id} 
                proto={mockObj}
                wrong={mockArray}
                reset={mockFalse}
            />
        )
    })

    it('should match the snap shot', () => {
        expect(wrapper).toMatchSnapshot()
    });

    it('should have defaut state', () => {
        expect(wrapper.state()).toEqual({showDescription: false, disableCard: mockFalse})
    })

    it('should check the type of the card against the value of the button', () => {
        wrapper.instance().checkType()
    })

    it('should check the type of the card against the value of the button', () => {
        wrapper.instance().checkType()
    })
})