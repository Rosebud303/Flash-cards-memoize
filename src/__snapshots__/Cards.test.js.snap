// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cards should match the snap shot 1`] = `
<input
  className="unclicked"
  onClick={[Function]}
/>
`;
